usage/rest-server/src/main/java/brooklyn/rest/security/provider/DelegatingSecurityProvider.java

```diff
@@ -94,10 +97,16 @@ private synchronized SecurityProvider loadDelegate() {
                 delegate = constructor.newInstance(mgmt);
             } catch (Exception e) {
                 constructor = clazz.getConstructor();
-                delegate = constructor.newInstance();
+                Object delegateO = constructor.newInstance();
+                if (!(delegateO instanceof SecurityProvider)) {
+                    // if classloaders get mangled it will be a different CL's SecurityProvider
+                    throw new ClassCastException("Delegate is either not a security provider or has an incompatible classloader: "+delegateO);
```

Comment:
I can't see any reason for it.  Makes sense to me to remove all the `target/test-rest-server` references, including in the pom.

It seems to work with them removed.